third_party/amd/backend/driver.py

```diff
@@ -163,14 +163,29 @@ def ty_to_cpp(ty):
         "u16": "uint16_t",
         "u32": "uint32_t",
         "u64": "uint64_t",
-        "fp16": "float",
-        "bf16": "float",
-        "fp32": "float",
-        "f32": "float",
+        "fp16": "double",
+        "bf16": "double",
+        "fp32": "double",
+        "f32": "double",
```

Comment:
why do we need double in those cases?

Reply:
Python's built-in `float` type are internally stored as 64-bit IEEE double. By having `double` here, we aren't losing any accuracy from Python to C, and all the conversion is done by later packing functions.

In terms of code, `PyArg_ParseTuple` parse Python float to C double according the `format` variable, which is derived from the `format_of` function, that looks up the dictionary in `ty_to_cpp`

https://github.com/triton-lang/triton/blob/973461ca22288e7bef5c981a47b01e63630e1b90/third_party/amd/backend/driver.py#L454

https://github.com/triton-lang/triton/blob/973461ca22288e7bef5c981a47b01e63630e1b90/third_party/amd/backend/driver.py#L244-L245

https://github.com/triton-lang/triton/blob/973461ca22288e7bef5c981a47b01e63630e1b90/third_party/amd/backend/driver.py#L230-L240